packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -215,7 +215,7 @@ class _CupertinoPickerState extends State<CupertinoPicker> {
     if (widget.backgroundColor != null && widget.backgroundColor.alpha < 255)
       return Container();
 
-    final Color widgetBackgroundColor = widget.backgroundColor;
+    final Color widgetBackgroundColor = widget.backgroundColor ?? const Color(0xFFFFFFFF);
```

Comment:
Instead of returning empty containers in various places, just don't add _buildGradientScreen and _buildMagnifierScreen to the stack when backgroundColor is null. This way, you can add asserts to _buildGradientScreen, _buildMagnifierScreen and _buildUnderMagnifierScreen too so the null handling isn't everywhere.